### PR TITLE
jmap_contact: map X-ABADR property to address.countryCode

### PR DESCRIPTION
**Pull request by rsto**
As it says on the tin.

Tested in https://github.com/cyrusimap/cassandane/tree/jmap_contact_xabadr

### CODE REVIEW

File: imap/jmap_contact.c [hunk 1]
```diff
@@ -3508,13 +3532,17 @@ static int _addresses_to_card(struct vparse_card *card,
...
             /* Add Apple-style label using property grouping */
             buf_reset(&buf);
             buf_printf(&buf, "adr%d", i);
             group = buf_cstring(&buf);
 
-            vparse_add_entry(card, group, VCARD_APPLE_LABEL_PROPERTY, label);
+            if (label)
+                vparse_add_entry(card, group, VCARD_APPLE_LABEL_PROPERTY, label);
+            if (countrycode)
+                vparse_add_entry(card, group, "X-ABADR", countrycode);
```
Comment (ksmurchison) [inline imap/jmap_contact.c:3544]: Do we want to create a macro for "X-ABADR" ?

**Review by ksmurchison — APPROVED** [review]
Other than my one nit, this looks good

**Review by brong — APPROVED** [review]
Yeah, this looks great - thanks :)

**Comment by brong** [comment]
This didn't get merged, and now breaks :(  I can't easily work out how to merge it, so I'm dropping it for now.

**Pull request merged**